src/inference/src/streams_executor.hpp

```diff
@@ -15,22 +15,40 @@ namespace ov {
 
 #ifdef __linux__
 /**
- * @brief      Parse processors infomation on Linux
+ * @brief      Parse cache based processors infomation on Linux
  * @ingroup    ie_dev_api_system_conf
  * @param[in]  _processors total number for processors in system.
- * @param[in]  _system_info_table system information for this platform.
+ * @param[in]  _system_info_table cache based system information for this platform.
  * @param[out] _sockets total number for sockets in system
  * @param[out] _cores total number for physical CPU cores in system
  * @param[out] _proc_type_table summary table of number of processors per type
  * @param[out] _cpu_mapping_table CPU mapping table for each processor
  * @return
  */
-void parse_processor_info_linux(const int _processors,
-                                const std::vector<std::vector<std::string>> _system_info_table,
-                                int& _sockets,
-                                int& _cores,
-                                std::vector<std::vector<int>>& _proc_type_table,
-                                std::vector<std::vector<int>>& _cpu_mapping_table);
+void parse_cache_info_linux(const int _processors,
+                            const std::vector<std::vector<std::string>> _system_info_table,
+                            int& _sockets,
+                            int& _cores,
+                            std::vector<std::vector<int>>& _proc_type_table,
+                            std::vector<std::vector<int>>& _cpu_mapping_table);
```

Comment:
should we create a struct / class which represents current CPU and use it instead of plain functions?
having 4 values returned by reference seems like a bad design